### PR TITLE
Rebuild and merge context menu code

### LINKED ISSUE #2747: Code base: Merge context menus so they can be reused in different layouts
**Is your feature request related to a problem? Please describe.**
We currently reimplement the right click context menu in every layout. This leads to duplicate code and can make the context menu a little harder to work with.

**Describe the solution you'd like**
The context menus that are reused can be merged so that changes will only need to be made once.

### LINKED ISSUE #4331: Create new file of type doesn't add specified extension
**Explain the bug**
I noticed the create new file no longer works with the type parameter.

**To reproduce**
1. Go to Folder
2. Click new -> Word doc or something
3. Enter name
4. File is not a word doc

**Expected behavior**
The file extension should be added.

**Logfile**
Nothing related.

**Screenshots**

https://user-images.githubusercontent.com/59544401/112729121-cf36d980-8ee7-11eb-9fde-78b453e24da5.mp4

**System Information**
 - OS Version: 21343.1000
 - App version: 1.5.2.0

**Additional comment**
Add any other comment about the problem here. Does this problem happen again after restarting the app?

### PR DESCRIPTION
**Pull request by winston-de**
**Resolved / Related Issues**

Itemize resolved / related issues by this PR.
- Closes #2747
 
**Details of Changes**
This is still a work in progress.

#### Goals:
- Separate the context menu items from the MenuFlyout control (and UI in general) to make it easier to implement #2342 (and possibly #2780) in the future.
- Add code to allow for customization of items to make it easier to implement #3966

## ToDo:
- [x] Remove the ".Text" from the string resource keys, and use the localized strings
- [x] Add recycle bin items
- [x] Add the layout flyout
- [x] Clean up/remove old context menu code in BaseLayout
- [ ] Add pin to start commands (depends on #4218)
- [x] Implement IsChecked for the model
- [x] Move items that are only shown on shift to the more menu if shift isn't held 
- [x] Closes #4331 

**Validation**

How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

**Screenshots (optional)**

Add screenshots here.

### CODE REVIEW

**Comment by winston-de** [comment]
Alright, I got most of the basic menu items implemented.

Without & with shift held.
image

**Comment by yaichenbaum** [comment]
@winston-de What still needs to be done?

**Comment by winston-de** [comment]
@yair100 I still have to add support custom font families for the glyphs (which is why the shortcut icon is wrong).
Also
- [ ] Remove the ".Text" from the string resource keys
- [ ] Add recycle bin items
- [ ] Add the base layout flyout
- [ ] Clean up/remove old context menu code in BaseLayout

**Comment by yaichenbaum** [comment]
@winston-de Sounds like good progress, this will make maintaining the context menu so much easier!

**Comment by winston-de** [comment]
@yair100 I've updated all the localization keys (removed the ".Text"). Should I run the multilingual app toolkit?

**Comment by winston-de** [comment]
I don't know what do do about the build error. It worked fine before I merged with main.
All it tells me is ```GENERATEPROJECTPRIFILE(0,0): Error : PRI175: 0x80073b0f - Processing Resources failed with error: Duplicate Entry. ```. 
I've tried turning up verbose logging to the "diagnostic" level, but it still doesn't show anything about where this is happening.
I've tried manually looking through each of the resource files changed in the merge for duplicates.
If anyone can help me debug this, it would be greatly appreciated.

**Comment by winston-de** [comment]
It looks like the build blows up (the error in the previous comment) when I remove the ".Text" from the string resources, though I'm not really sure why. The only option I see is to create new ones, but that would mean loosing all the current translations.

**Comment by winston-de** [comment]
Yay it worked!

**Comment by duke7553** [comment]
@winston-de Great! Once this is finished, we can finally start on the CommandbarFlyout update for the newly unified flyout.

**Comment by winston-de** [comment]
> @winston-de Great! Once this is finished, we can finally start on the CommandbarFlyout update for the newly unified flyout.

I did a little bit of testing, it should be a relatively simple change as well.

**Comment by winston-de** [comment]
I've been noticing that there's been some issues with users not being able to find items that are hidden unless you shift+right click. What do you think of moving those items to the "more" menu when shift isn't held?

It would look something like this:

**Comment by winston-de** [comment]
Inadvertently, this also mostly fixed #4331. I'll fix the nav toolbar one as well and just bundle the full fix in this pr.

File: Files/BaseLayout.cs
Comment (duke7553) [inline Files/BaseLayout.cs:431]: Should these events be unsubscribed in the OnNavigatingFrom() override?

**Review by lukeblevins — DISMISSED** [review]
LGTM

**Comment by winston-de** [comment]
I think we should wait for #4218 to be merged before merging this one, as it adds a command to pin and unpin items to the Start menu. Currently pin to start doesn't do anything since there isn't a command.

**Comment by yair100** [comment]
> @yair100 I've updated all the localization keys (removed the ".Text"). Should I run the multilingual app toolkit?

@winston-de Sure! By the way, why do a lot of the keys have the number 2 attached to them?

> I've been noticing that there's been some issues with users not being able to find items that are hidden unless you shift+right click. What do you think of moving those items to the "more" menu when shift isn't held?

This sounds like a good idea!

**Comment by winston-de** [comment]
> > @yair100 I've updated all the localization keys (removed the ".Text"). Should I run the multilingual app toolkit?
> 
> @winston-de Sure! By the way, why do a lot of the keys have the number 2 attached to them?

@yair100 I don't know why, but it wouldn't build unless I added them.

**Comment by yaichenbaum** [comment]
@winston-de Have you checked to make sure there are no duplicate keys?

**Comment by winston-de** [comment]
@yair100 Yes, there weren't any. I think the problem initially was that I was not changing all the keys, just then ones in the en-us file, so the compiler saw that as duplicate keys. Now that I'm changing them all, I can probably remove the 2s without any issue.

**Comment by winston-de** [comment]
Ok, so I tried removing the 2 for one set of resources, and it built just fine. Then I removed them all, and the build failed with the duplicate error. I checked each resource key for duplicates, and didn't find any. I'm guessing one of them is causing the problem for whatever, but I have no clue which one (except that it's not BaseLayoutContextFlyoutLayoutMode).

**Comment by yaichenbaum** [comment]
@winston-de Another possibility is that one of the resources are being used where they shouldn't and removing the `.Text` is throwing the error. The best solution I can think of for now is to go through each string one at a time.

**Comment by winston-de** [comment]
@yair100 That would take me hours to do, are the 2s really that bad? It works fine as it is. And if so, is it possible that I could come back and fix them in their own PR?

**Comment by winston-de** [comment]
Also, @TyJOrtiz is it alright with you that I make the necessary changes to implement the new context menus in Column View?

**Comment by TyJOrtiz** [comment]
@winston-de Yes that's fine

**Comment by yair100** [comment]
> @yair100 That would take me hours to do, are the 2s really that bad? It works fine as it is. And if so, is it possible that I could come back and fix them in their own PR?

I don't want to cause all that extra work, how about keeping the new keys, but can you change the names of the new keys to something like `RegularContextFlyout...`? This will make it easier to search for the older keys and remove them in a future PR.

**Comment by winston-de** [comment]
@yair100 I can do that, but I don't see why you would to search through and remove the old keys, as they're gone. I changed the keys with find/replace throughout the entire solution so that the current translations wouldn't be lost.

**Comment by winston-de** [comment]
The new context menus in Column View are up and running. 
image

Btw, @TyJOrtiz the column view is awesome! :)

**Comment by yair100** [comment]
> @yair100 I can do that, but I don't see why you would to search through and remove the old keys, as they're gone. I changed the keys with find/replace throughout the entire solution so that the current translations wouldn't be lost.

There is a duplicate key somewhere, I will try to find it.

**Comment by yaichenbaum** [comment]
@winston-de It seems like the sort by "Date deleted" option is showing in all directories, additionally, the layout menu items seem to be disabled.

**Comment by winston-de** [comment]
> > @yair100 I can do that, but I don't see why you would to search through and remove the old keys, as they're gone. I changed the keys with find/replace throughout the entire solution so that the current translations wouldn't be lost.
> 
> There is a duplicate key somewhere, I will try to find it.

@yair100 Thanks for doing this.
I searched every key for anything with the same name throughout the entire solution, and each one had somewhere between 37-54 references (which corresponds to the number of languages it was translated to). If there is a duplicate, I think it exists in only one resource file. I don't know if that's of any help to you.

**Review by yair100 — APPROVED** [review]

**Comment by winston-de** [comment]
@yair100 It looks like I may have never had to remove the ".Text" from everything.
I didn't realize it was possible to access them by using a slash, like this ```"resourcename/Text".GetLocalized()```.
I can change them all back if you'd prefer.

**Comment by winston-de** [comment]
@yair100 I apologize for this, I just learned that the resources could be accessed like that.

**Pull request merged**
